azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1827,6 +1834,7 @@ private void processQueuedFlows(long activeExecutorsRefreshWindow,
         if(exflow.getUpdateTime() > lastExecutorRefreshTime) {
           // put back in the queue
           queuedFlows.enqueue(exflow, reference);
+          runningCandidate = null;
```

Comment:
@suvodeep-pyne sorry I am supposed to put it in the commit message, but will do it when merging. So there're two threads: producer thread which puts flow to be executed into the waiting queue, and consumer thread which polls from the queue and dispatch for execution. There' a short gap between consumer polls an flow(runningCandidate in the code) from the waiting queue and dispatching it, during which the flow is neither in waiting queue and running queue. At that time, producer checks the same flow and finds that it's not in any of the queue, assuming it's not running, then puts it in the waiting queue. Then two same flow are going to be dispatched. The fix is making the runningCandidate as shared variable so that it can be checked by producer as well.